Project21/Project21/conversionfile.c

```diff
@@ -0,0 +1,36 @@
+// ------------------------------------------------------------------------------------------------
+// Assignment 3 ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.Naveenkumar s
+// ------------------------------------------------------------------
+// Assignment 3
+// Decimal to Binary conversion and Decimal to HexaDecimal conversion
+// ------------------------------------------------------------------------------------------------
+#define INT_SIZE sizeof(int)*8
+#define HEX_SIZE 8
+
+char binary[INT_SIZE + 1];
+char hex[HEX_SIZE + 1];
+
+/// <summary>Converts a decimal number to binary</summary>
+char* DecimalToBinary (int n) {

```

Comment:
It is also advised to not return local pointers from a function as it causes multiple bugs. It is better to pass a pointer declared in the caller function and assign the return value to it to prevent these.